consensus/src/marshal/actor.rs

```diff
@@ -0,0 +1,850 @@
+use super::{
+    config::Config,
+    finalizer::Finalizer,
+    handler::{self, Handler},
+    ingress::{Mailbox, Message, Orchestration, Orchestrator},
+    types::{Block, Finalized, Notarized},
+};
+use crate::threshold_simplex::types::Finalization;
+use commonware_broadcast::{buffered, Broadcaster};
+use commonware_codec::{DecodeExt, Encode};
+use commonware_cryptography::{
+    bls12381::primitives::variant::Variant, Digest, Digestible, PublicKey,
+};
+use commonware_macros::select;
+use commonware_p2p::{utils::requester, Receiver, Recipients, Sender};
+use commonware_resolver::{
+    p2p::{self, Coordinator},
+    Resolver,
+};
+use commonware_runtime::{Clock, Handle, Metrics, Spawner, Storage};
+use commonware_storage::{
+    archive::{self, immutable, prunable, Archive as _, Identifier},
+    metadata::{self, Metadata},
+    translator::TwoCap,
+};
+use commonware_utils::array::{FixedBytes, U64};
+use futures::{channel::mpsc, try_join, StreamExt};
+use governor::{clock::Clock as GClock, Quota};
+use prometheus_client::metrics::gauge::Gauge;
+use rand::Rng;
+use std::{
+    collections::BTreeSet,
+    marker::PhantomData,
+    time::{Duration, Instant},
+};
+use tracing::{debug, info, warn};
+
+/// Application actor.
+pub struct Actor<
+    D: Digest,
+    R: Rng + Spawner + Metrics + Clock + GClock + Storage,
+    V: Variant,
+    P: PublicKey,
+    Z: Coordinator<PublicKey = P>,
+> {
+    context: R,
+    public_key: P,
+    identity: V::Public,
+    coordinator: Z,
+    mailbox: mpsc::Receiver<Message<V, D>>,
+    mailbox_size: usize,
+    backfill_quota: Quota,
+    namespace: Vec<u8>,
+
+    // Blocks verified stored by view<>digest
+    verified: prunable::Archive<TwoCap, R, D, Block<D>>,
+    // Blocks notarized stored by view<>digest
+    notarized: prunable::Archive<TwoCap, R, D, Notarized<V, D>>,
+
+    // Finalizations stored by height
+    finalized: immutable::Archive<R, D, Finalization<V, D>>,
+    // Blocks finalized stored by height
+    //
+    // We store this separately because we may not have the finalization for a block
+    blocks: immutable::Archive<R, D, Block<D>>,
```

Comment:
As discussed, I think we should only ever need to use the `prunable::Archive` in here (just need to ensure we keep stuff long enough for the application to ACK)